Fix missing reaction and edit buttons on permissioned channels

jrainville opened the pull request:
Fixes #14089

This fix is partly a refactor.

The problem was that we relied on permissions to show the buttons, but that is not the right approach.

The CommunityChat gives us all that we need to know already with `canPost`, `canView` and `canPostReactions`. We should never be reliant on permissions, because it's way more complex and can break, like it did here (probably with the refactor to not check the permissions ourselves).

So while this PR is a fix, is also introduces the very important properties `canPost` and `canVierw` to the ChatDto and the ChatItem. Do note that those properties only come from the CommunityChat, not the chatService's Chats.

We should start refactoring some more of the code to use those instead of the permissions.

**Known issue that should be fixed on master:** when a permission is edited, we recalculate the permissions and something goes wrong and it shows the blurred screen. You need to change channel and come back for it to look ok again.
Again, if we stop to check permissions when we already have access to the canPost and canView properties, we will be able to fix the issue.

status-im-auto commented:
### Jenkins Builds
| :grey_question: | Commit | :hash: | Finished (UTC) | Duration | Platform | Result |
|-|-|-|-|-|-|-|
| :heavy_check_mark: | e85b6777 | [#1](https://ci.status.im/job/status-desktop/job/prs/job/linux/job/x86_64/job/tests-nim/job/PR-14933/1/) | 2024-05-27 21:04:28 | ~7 min | `tests/nim` | [:page_facing_up:`log`](https://ci.status.im/job/status-desktop/job/prs/job/linux/job/x86_64/job/tests-nim/job/PR-14933/1//consoleText) |

jrainville commented:
Turns out I can still reproduce the issue on this branch. I'll continue working on it

jrainville commented:
Ok, I fixed it. The issue was that we had the most error prone confusing code ever in chat_section module (I'm pretty sure I added it too :confounded: )

That code made it so that almost all calls  to `addOrUpdateChat` contained an empty CommunityDto, meaning that all properties were reset to `false.`

endulab commented on src/app/modules/main/chat_section/chat_content/controller.nim:
```diff
@@ -184,12 +184,6 @@ proc init*(self: Controller) =
-    var args = ChatUpdateArgs(e)
-    for chat in args.chats:
-      if self.chatId == chat.id:
-        self.delegate.onChatEdited(chat)
```
why this signal doesn't need to be handled here?

jrainville replied:
I made it so that it's the parent (`chat_section`) that calls `onChatUpdated` instead. That way, the `chat_section` module can pass us the right chatItem. t reduces the duplication of code.

endulab commented:
Good job!

jrainville commented:
@friofry @endulab can you review please

jrainville commented on src/app/modules/main/chat_section/chat_content/chat_details.nim:
```diff
@@ -248,6 +254,28 @@ QtObject:
     self.blocked = value
     self.blockedChanged()
 
+  proc canPostChanged(self: ChatDetails) {.signal.}
```
That would be great indeed!
